jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -0,0 +1,286 @@
+   *
+   * @param classFilesDirectory the directory containing the class files
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addClasses(Path classFilesDirectory) throws IOException {
+    if (!Files.exists(classFilesDirectory)) {
+      throw new IOException("Directory '" + classFilesDirectory + "' does not exist.");
+    }
+    if (!Files.isDirectory(classFilesDirectory)) {
+      throw new IOException(
+          "Adding classes failed: '" + classFilesDirectory + "' is not a directory");
+    }
+    layerConfigurationsBuilder.addDirectoryContents(
+        LayerType.CLASSES, classFilesDirectory, path -> true, CLASSES_PATH);
+    classpath.add(CLASSES_PATH.toString());
+    return this;
+  }
+
+  /**
+   * Adds additional files to the classpath.
+   *
+   * @param otherFiles the list of files to add. Files are added to {@code /app/other} on the
+   *     container file system. If {@code otherFiles} contains a directory, files within are added
+   *     recursively.
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addToClasspath(List<Path> otherFiles) throws IOException {
+    for (Path file : otherFiles) {
+      if (!Files.exists(file)) {
+        throw new IOException("File '" + file + "' does not exist.");
+      }
+      if (Files.isDirectory(file)) {
+        layerConfigurationsBuilder.addDirectoryContents(
+            LayerType.EXTRA_FILES, file, path -> true, OTHERS_PATH);
+      } else {
+        layerConfigurationsBuilder.addFile(
+            LayerType.EXTRA_FILES, file, OTHERS_PATH.resolve(file.getFileName()));
+      }
+    }
+    classpath.add(OTHERS_PATH.toString());
+    return this;
+  }
+
+  /**
+   * Adds additional files to the classpath.
+   *
+   * @param otherFiles the list of files to add. Files are added to /app/other on the container file
+   *     system. If {@code otherFiles} contains a directory, files within are added recursively.
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addToClasspath(Path... otherFiles) throws IOException {
+    return addToClasspath(Arrays.asList(otherFiles));
+  }
+
+  /**
+   * Adds JVM flags to use when starting the application.
+   *
+   * @param jvmFlags the list of JVM flags to add
+   * @return this
+   */
+  public JavaContainerBuilder addJvmFlags(List<String> jvmFlags) {
+    this.jvmFlags.addAll(jvmFlags);
+    return this;
+  }
+
+  /**
+   * Adds JVM flags to use when starting the application.
+   *
+   * @param jvmFlags the list of JVM flags to add
+   * @return this
+   */
+  public JavaContainerBuilder addJvmFlags(String... jvmFlags) {
+    this.jvmFlags.addAll(Arrays.asList(jvmFlags));
+    return this;
+  }
+
+  /**
+   * Sets the main class used to start the application on the image. To automatically infer the main
```

Comment:
nit: "automatically infer" doesn't really apply here - "To find the main class from {@code .class} files, use..."